retrofit2/src/main/java/com/linecorp/armeria/client/retrofit2/InvocationUtil.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.retrofit2;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.logging.RequestLog;
+
+import io.netty.util.AttributeKey;
+import retrofit2.Invocation;
+
+/**
+ * Armeria Retrofit automatically adds an invocation to {@link RequestLog}. You can retrieve
+ * the invocation by {@link InvocationUtil#getInvocation(RequestLog)} for metrics and monitoring.
+ *
+ * @see Invocation
+ */
+public final class InvocationUtil {
```

Comment:
> I'm not sure I like the name, but I have a better idea either. Any other suggestions?

How about removing this class and moving the logic into the existing classes?
We can place `RETROFIT_INVOCATION` into `ArmeriaCallFactory` I think. `getInvocation` is used only once and `setInvocation` is used twice but we can just use it only once by removing the duplicate code. If we do that, we don't even have to use try with resources when the `invocation` is `null`.